src/compiler/emitter.ts

```diff
@@ -731,7 +737,11 @@ namespace ts {
         }
 
         function emitBodyIndirect(node: Node, elements: NodeArray<Node>, emitCallback: (node: Node) => void): void {
-            if (emitBodyWithDetachedComments) {
+            // If the node is a sourceFile and it has prologueDirective (statmentOffSet is not zero) that is synthesize
+            // We will need to emit detached comment here because emitPrologueDirective will not emit comments of such prologue directive
+            const shouldEmitDetachedComment = node.kind !== SyntaxKind.SourceFile ? true :

```

Comment:
Why not just perform this check in `emitSourceFile` rather than pay the cost for the check in every function body?